Bundle bulk operation on web ui

yipenghe opened the pull request:
Implementing #1064 
> Supports bulk bundle operation from the web UI
> Can use 'd' for detach, 'backspace/del' for delete, 'v' for kill on multiple selected bundles, they open up a dialog for confirmation, can 'f' to select force deletion on deletion dialog, can use 'enter' to confirm action. It's the same as running cl rm, cl detach & cl kill on selected bundles
> show/hides bulk bundle action bar when at least one/none is selected, hides the worksheet action bar when bundles are selected (can still use shortcuts to perform)
> Most shortcuts are disabled when a dialog is opened
> Currently the checkmark get refreshed when a new bundle row is added / a bundle operation is successfully performed, the selection persists if bundle operation failed
> some minor changes to bundle row style

yipenghe committed:
bundle bulk operation draft

yipenghe committed:
bundle bulk operation draft

yipenghe committed:
delete & kill, remove check after operation

yipenghe committed:
delete & kill, remove check after operation

yipenghe committed:
null run for attach&detach

yipenghe committed:
null run for attach&detach

yipenghe committed:
support force delete in bulk operation

yipenghe committed:
support force delete in bulk operation

yipenghe committed:
add error dialog when action fails

yipenghe committed:
add error dialog when action fails

yipenghe committed:
No bundle row selected dialog

yipenghe committed:
No bundle row selected dialog

epicfaace commented:
When deleting (or bulk deleting) a bundle, show an extra dialog (warning -- "these bundles are attached to more than 1 worksheet. are you sure you want to delete --force?" if force delete is needed

epicfaace commented:
- Checkbox in the header to "select all"
- keyboard shortcuts (could do later)

epicfaace commented:
Change "bundles" dropdown to show the four buttons; only show them when at least one checkbox is selected.

yipenghe committed:
shallow change to use button instead of menu

yipenghe committed:
shallow change to use button instead of menu

yipenghe committed:
show/hide bundle operations when at least one/no bundle is selected

yipenghe committed:
show/hide bundle operations when at least one/no bundle is selected

yipenghe committed:
mimic gmail behavior

yipenghe committed:
mimic gmail behavior

yipenghe committed:
minor fix

yipenghe committed:
minor fix

yipenghe committed:
fix recreating bundles dictionary every time in Worksheet.js

yipenghe committed:
fix recreating bundles dictionary every time in Worksheet.js

yipenghe committed:
remain selected after action

yipenghe committed:
remain selected after action

yipenghe committed:
Default check newly added bundleRows if select all is checked

yipenghe committed:
Default check newly added bundleRows if select all is checked

yipenghe committed:
shouldupdate

yipenghe committed:
shouldupdate

yipenghe committed:
checkbox alignment

yipenghe committed:
checkbox alignment

yipenghe committed:
indeterminate icon size with SVG Icon

yipenghe committed:
indeterminate icon size with SVG Icon

yipenghe committed:
address one-uuid-multiple-possible-rows

yipenghe committed:
address one-uuid-multiple-possible-rows

yipenghe committed:
remain selected after operation except for sucessful deletion

yipenghe committed:
remain selected after operation except for sucessful deletion

yipenghe commented:
The selection logic should be working now, following the behaviours of gmail style...bulk delete & kill are working
Remaining: 1. Force delete logic 2. attach/detach command 3. style the four buttons

Levilian committed:
Add keyboard shortcuts for major actions (#1578)

* Add keyboard shortcuts for major actions
- insert text after current cell
- upload after current cell
- run after current cell
- enter for markdown editing
- ctrl+enter for saving
- backspace to delete
- fixed moving to beginning shortcut
- fixed refreshing worksheet focus

* fixed focus and shortcuts issues
- added del key to delete
- added esc to go back
- fixed focus bug on pressing keys in terminal
- added keyboard shortcut to focus the terminal
- implemented re-focus on delete, run, cell, and upload

yipenghe committed:
merge

yipenghe committed:
hide button bar after deletion

yipenghe committed:
hide button bar after deletion

yipenghe committed:
use unique identifier for mapping, remove child key in tableitem after deletion

yipenghe committed:
merge

yipenghe commented:
Some issues:
When select ‘Delete’, want to be able to hit ENTER to confirm, but currently, this triggers something in the back

Message ‘Failed to perform this actions’ shows up in a popup box but unclear how to close it (should have an ‘x’ in the corner)

Force delete - should have an [i] with a tooltip that explains what this is

yipenghe committed:
temp

yipenghe committed:
Merge branch 'frontend' into f-web-bulk

yipenghe committed:
add worksheet argument

yipenghe committed:
Disabling attach for now

yipenghe committed:
bind enter to confirm actions

yipenghe committed:
dont trigger things behind dialog, temp solution

yipenghe committed:
Merge branch 'frontend' into f-web-bulk (rebase)

yipenghe committed:
add dismiss button for error messages

yipenghe committed:
More information about force deletion

yipenghe committed:
More information about force deletion

yipenghe committed:
Merge branch 'frontend' into f-web-bulk

yipenghe committed:
remove out-of-date comments

yipenghe committed:
remove floating deletion on bundle row

yipenghe committed:
remove attach button

yipenghe committed:
layout & shortcuts, disable worksheet/bundlerow shortcuts when a dialog is open, use x to select a bundle row, use space to select all rows in a table

yipenghe committed:
fix key error

yipenghe committed:
update tableitem control bundlerow checkstatus logic, fix deletion bug, added comments to document the mechanism

yipenghe committed:
remove unused code
